tardis/model/radiation_field_state.py

```diff
@@ -17,18 +17,37 @@ class DiluteBlackBodyRadiationFieldState:
         Radiative temperature in each shell
     dilution_factor : numpy.ndarray
         Dilution Factors in each shell
+    geometry: tardis.model.Radial1DModel
+        The geometry of the model that uses to constrains the active shells
     """
 
     def __init__(
         self,
         t_radiative: u.Quantity,
         dilution_factor: np.ndarray,
+        geometry=None,
     ):
         # ensuring that the radiation_field has both
         # dilution_factor and t_radiative equal length
         assert len(t_radiative) == len(dilution_factor)
-        assert np.all(t_radiative > 0 * u.K)
-        assert np.all(dilution_factor > 0)
+        if (
+            geometry is not None
+        ):  # check the active shells only (this is used when setting up the radiation_field_state)
+            assert np.all(
+                t_radiative[
```

Comment:
That is actually already being taken of, since the estimators only use the active shells, the simulation_state slice the active shells from the "raw" radiation_field_state.